### DISCUSSION TITLE
How to configure executors in allure report

### DISCUSSION THREAD

**Discussion by Krishna-Vasarla** [allure-framework/allure2#2249] [Questions & Support] [answered]
I tried adding the executors.json file in the allure-results while generating the report it was throwing an error.
I also tried  allure generate --config ./allure-config.json --clean where in config.json file I added the executors. But it also throwing an error like "Could not find config file .\allure-config.json. Using the empty configuration"
Could you please help me

**Comment by baev** [comment, marked as answer]
The correct file name is `executor.json`. The only option to specify executor data is by adding the `executor.json` file into `allure-results`.

Sample content:

```json
{
  "name": "Jenkins",
  "type": "jenkins",
  "url": "http://example.org",
  "buildOrder": 13,
  "buildName": "allure-report_deploy#13",
  "buildUrl": "http://example.org/build#13",
  "reportUrl": "http://example.org/build#13/AllureReport",
  "reportName": "Demo allure report"
}
```

PS created a task to add the information about `executor.json` to https://allurereport.org/docs/

**Reply by harmin-parra** [reply]
Unfortunately, the line  **"reportName": "Demo allure report"** doesn't affect in any way the final Report.
What is the purpose of the **reportName** key?

**Reply by harmin-parra** [reply]
What is the purpose of the **reportUrl** key?

**Reply by baev** [reply]
> Unfortunately, the line "reportName": "Demo allure report" doesn't affect in any way the final Report.
What is the purpose of the reportName key?

fixed via https://github.com/allure-framework/allure2/pull/2229

**Reply by baev** [reply]
> What is the purpose of the reportUrl key?

To make trend/history clickable

**Comment by anci3ntr0ck** [comment]
@harmin-parra After reading the source code of `allure-generator` I found this snippet in `*TrendPlugin`s: 
```java
        extractLatestExecutor(launchesResults).ifPresent(info -> {
            item.setBuildOrder(info.getBuildOrder());
            item.setReportName(info.getReportName());
            item.setReportUrl(info.getReportUrl());
        });
```
which means get `buildOrder`, `reportName` and `reportUrl` from the lastest `executor.json` file (which has the biggest `buildOrder`) in result dirs,  and set them in `*-trend.json` files. Then you can see the `buildOrder` on trend graphs:
![image](https://github.com/allure-framework/allure2/assets/55662947/3fadfd50-e610-4d60-ad8a-0824740b1dd9)

> What is the purpose of the reportUrl key?

Click the red box area in the image above, then you can jump to the corresponding report url.

> What is the purpose of the reportName key?

Seems no use anywhere for now, the `reportName` is only used in `widget-summary` in the whole allure2 project
![image](https://github.com/allure-framework/allure2/assets/55662947/8c7dcb1c-bfba-435b-a61e-92fd8ba6b9be)
![image](https://github.com/allure-framework/allure2/assets/55662947/cb96a8e2-eee8-47eb-bef1-efd97a6d93eb)
But the `reportName` in `summary.json` is **ROUGHLY** set to a fixed value, you can only change it in report dir after the report is generated 😫
```java
public class SummaryPlugin {
    ...
   
    @Override
    protected SummaryData getData(final List<LaunchResults> launches) {
        final SummaryData data = new SummaryData()
                .setStatistic(new Statistic())
                .setTime(new GroupTime())
                .setReportName("Allure Report");
        launches.stream()
                .flatMap(launch -> launch.getResults().stream())
                .forEach(result -> {
                    data.getStatistic().update(result);
                    data.getTime().update(result);
                });
        return data;
    }
}
```
@baev How about using `reportName` in `executor.json`  for the summary, just like what `*TrendPlugin`s do?

**Reply by baev** [reply]
fixed via https://github.com/allure-framework/allure2/pull/2229
